src/server/pps/server/api_server.go

```diff
@@ -2492,6 +2492,13 @@ func (a *apiServer) initializePipelineInfo(txn *pps.CreatePipelineTransaction, o
 		if !request.Reprocess {
 			pipelineInfo.Details.Salt = oldPipelineInfo.Details.Salt
 		}
+		pipelineInfo.Details.CreatedBy = oldPipelineInfo.Details.CreatedBy
+	} else {
+		if whoAmI, err := txnCtx.WhoAmI(); err == nil {
```

Comment:
I would just pass the username to this function rather than the entire txnCtx.  Remember that PPS transactions are deferred; stored in the database until commit time.  This means that the principal might not exist in principals or auth_tokens by the time the tx actually executes.

Also, txnCtx.WhoAmI() cannot return any error other than ErrNotActivated, so it's not worth checking.  I'm still waiting on a review for a PR that adds txnCtx.Username() that can't error.  I'd recommend using that.

Reply:
The principals table is append-only today, but that can change in the future, and that change would break transactions.  So I think it would be good to authdb.EnsurePrincipal at commit time, rather than at transaction creation time.

Reply:
In this case it won’t though, since pipelines are non-relational. The created-by field is just a string in the protobuf bytes.

Reply:
Don't we want to ensure that the data is consistent so the eventual migration doesn't have to handle that?